pkg/controller/daemonset/daemonset_controller.go

```diff
@@ -390,6 +403,36 @@ func (dsc *ReconcileDaemonSet) syncDaemonSet(request reconcile.Request) error {
 		return dsc.updateDaemonSetStatus(ds, nodeList, hash, false)
 	}
 
+	if !isPreDownloadDisabled && dsc.Client != nil {
+		if ds.Status.UpdatedNumberScheduled != ds.Status.DesiredNumberScheduled ||
+			hash != ds.Status.DaemonSetHash {
+			// get ads pre-download annotation
+			minUpdatedReadyPodsCount := 0
+			if minUpdatedReadyPods, ok := ds.Annotations[appsv1alpha1.ImagePreDownloadMinUpdatedReadyPods]; ok {
+				minUpdatedReadyPodsIntStr := intstrutil.Parse(minUpdatedReadyPods)
+				minUpdatedReadyPodsCount, err = intstrutil.GetScaledValueFromIntOrPercent(&minUpdatedReadyPodsIntStr, int(ds.Status.DesiredNumberScheduled), true)
+				if err != nil {
+					klog.Errorf("Failed to GetScaledValueFromIntOrPercent of minUpdatedReadyPods for %s: %v", request, err)
+				}
+			}
+			updatedReadyReplicas := ds.Status.UpdatedNumberScheduled
```

Comment:
UpdatedNumberScheduled may container replicas which are not ready, maybe we should add an extra status field such as updatedReady

Reply:
Yes,  you are right.
I will enable imagepredownload without checking updateReadyReplicas.